src/Hl7.Fhir.Core.Tests/Rest/TestResourceIdentifier.cs

```diff
@@ -249,7 +249,7 @@ public void TestIsResourceId()
             Assert.IsFalse(ResourceIdentity.IsRestResourceIdentity("http://nu.nl/fhir/crap"));
             Assert.IsFalse(ResourceIdentity.IsRestResourceIdentity("x/y/z/4"));
             Assert.IsFalse(ResourceIdentity.IsRestResourceIdentity("urn:oid:1.2.3.4.5"));
+            Assert.IsFalse(ResourceIdentity.IsRestResourceIdentity("XXX/_history/1"));
```

Comment:
Unfortunately, this unit test does not test the parsing. It only checks if the uri matches the pattern. 
Please add an new testmethod like `TestBuild`.